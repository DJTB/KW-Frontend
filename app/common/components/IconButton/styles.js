import styled from 'styled-components';

import { gutter } from 'common/styles/layout';
import { resetButton } from 'common/styles/utils';
import { fastEaseQuad } from 'common/styles/animation';

export const Button = styled.button`
  ${resetButton}
  ${gutter()}
  display: flex;
  justify-content: center;
  align-self: center;
  align-items: center;
  vertical-align: middle;
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;
  transform: scale(1);

  &:not(:disabled) {
    &:focus,
    &:hover {
      opacity: .9;
      outline: none;
    }

    &:active {
      opacity: 1;
      transform: scale(.9);
    }
  }

  &:disabled {
    cursor: not-allowed;
  }
`;
