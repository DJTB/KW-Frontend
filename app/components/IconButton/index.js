import React from "react";
import PropTypes from "prop-types";

import Icon from "components/Icon";
import Button from "base/Button";

// FIXME: just use <Button plainButton><Icon /></Button> instead..... geez

IconButton.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.oneOf(["submit", "reset", "button"]),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  inline: PropTypes.bool,
  children: PropTypes.node,
  plainButton: PropTypes.bool,
};

IconButton.defaultProps = {
  type: "button",
  plainButton: true,
  color: "currentColor",
  size: "1.5em",
  disabled: false,
  inline: false,
  children: null,
  onClick: (event) => event /* passthrough, for submit buttons in forms with onSubmit */,
};

function IconButton({
  name, title, color, size, inline, children, ...props
}) {
  return (
    <Button aria-label={title} {...props}>
      {children}
      <Icon name={name} inline={inline} size={size} color={color} />
    </Button>
  );
}

export default IconButton;
