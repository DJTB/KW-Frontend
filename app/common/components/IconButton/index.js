import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { gutter } from 'common/styles/layout';

import Aux from 'common/components/Aux';
import Button from 'common/components/Button';
import Icon from 'common/components/Icon';

// prettier-ignore
export const Text = styled.span`
  ${gutter({ position: 'left' })}
  ${gutter({ position: 'right', mod: 2 })}
`;

IconButton.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.oneOf(['submit', 'reset', 'button']),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  inline: PropTypes.bool,
  plainButton: PropTypes.bool,
  text: PropTypes.any,
  isSubmitting: PropTypes.bool,
};

IconButton.defaultProps = {
  type: 'button',
  plainButton: true,
  color: 'currentColor',
  size: '1.5em',
  disabled: false,
  inline: false,
  text: '',
  isSubmitting: false,
  onClick: (event) => event /* passthrough, for submit buttons in forms with onSubmit */,
};

function IconButton({ name, text, title, color, size, inline, isSubmitting, ...props }) {
  const RenderedText = text ? <Text>{text}</Text> : null;
  const RenderedIcon = (
    <Icon
      isRotating={isSubmitting}
      name={isSubmitting ? 'SYNC' : name}
      inline={inline}
      size={size}
      color={color}
    />
  );

  return (
    <Button {...props} title={title} aria-label={title} color={color}>
      <Aux>
        {RenderedText}
        {RenderedIcon}
      </Aux>
    </Button>
  );
}

export default IconButton;
