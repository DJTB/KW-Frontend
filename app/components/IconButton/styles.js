import styled from 'styled-components';

import { resetButton } from 'shared/styles/utils';
import { fastEaseQuad } from 'shared/styles/animation';

export const Button = styled.button`
  ${resetButton}
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;

  &:not(:disabled) {
    &:focus,
    &:hover {
      opacity: .9;
      outline: none;
    }

    &:active {
      opacity: 1;
      transform: scale(.9);
    }
  }

  &:disabled {
    cursor: not-allowed;
  }
`;
