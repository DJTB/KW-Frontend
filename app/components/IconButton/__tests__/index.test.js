import "jest-styled-components";
import React from "react";
import { render } from "enzyme";
import IconButton from "../index";

describe("<IconButton /> ", () => {
  it("should match baseline snapshot", () => {
    const renderedComponent = render(<IconButton name="ADD" title="Does an action" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it("should adopt button types", () => {
    const renderedComponent = render(
      <IconButton
        type="submit"
        name="ADD"
        title="Does an action"
        handleClick={jest.fn()}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
