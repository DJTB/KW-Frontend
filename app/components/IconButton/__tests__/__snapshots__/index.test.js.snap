// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton />  should adopt button types 1`] = `
.c1 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  width: 1.5em;
  height: 1.5em;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c2 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  fill: currentColor;
}

.c0 {
  padding: 0.225rem;
  font-size: 15.3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  min-width: 8em;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  color: #fafafa;
  border: 2px solid #4c81d6;
  background-color: #4c81d6;
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
}

.c0 {
  background: none;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  border-radius: 0;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
  -webkit-user-select: none !important;
  -moz-user-select: none !important;
  -ms-user-select: none !important;
  min-width: auto;
}

.c0::-moz-focus-inner {
  border: 0 !important;
  padding: 0 !important;
}

@media (min-width:600px) {
  .c0 {
    padding: 0.33325rem;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c0 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c0 {
    font-size: 17.4px;
  }
}

<button
  aria-label="Does an action"
  class="c0"
  color="#fafafa"
  type="submit"
>
  <span
    class="c1"
    color="currentColor"
  >
    <svg
      class="c2"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      title="ADD"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</button>
`;

exports[`<IconButton />  should match baseline snapshot 1`] = `
.c1 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  width: 1.5em;
  height: 1.5em;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c2 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  fill: currentColor;
}

.c0 {
  padding: 0.225rem;
  font-size: 15.3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  min-width: 8em;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  color: #fafafa;
  border: 2px solid #4c81d6;
  background-color: #4c81d6;
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
}

.c0 {
  background: none;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  border-radius: 0;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
  -webkit-user-select: none !important;
  -moz-user-select: none !important;
  -ms-user-select: none !important;
  min-width: auto;
}

.c0::-moz-focus-inner {
  border: 0 !important;
  padding: 0 !important;
}

@media (min-width:600px) {
  .c0 {
    padding: 0.33325rem;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c0 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c0 {
    font-size: 17.4px;
  }
}

<button
  aria-label="Does an action"
  class="c0"
  color="#fafafa"
  type="button"
>
  <span
    class="c1"
    color="currentColor"
  >
    <svg
      class="c2"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      title="ADD"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</button>
`;
